aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -300,10 +300,34 @@ The `SampleHealthCheckWithDiConfig` and the Health check needs to be added to th
 
 :::code language="csharp" source="~/host-and-deploy/health-checks/samples/7.x/HealthChecksSample/Snippets/Program.cs" id="snippet_MapHealthChecksUsingDependencyInjection":::
 
+## UseHealthChecks vs. MapHealthChecks
+
+The `MapHealthChecks` method accepts arguments similar to `UseHealthChecks`. The advantage of using `MapHealthChecks` over `UseHealthChecks` is the ability to apply authorization and to have greater fine-grained control over the matching policy. The primary advantage of using `UseHealthChecks` over `MapHealthChecks` over terminating the middleware and selecting endpoints via port.
+
+<xref:Microsoft.AspNetCore.Builder.HealthCheckApplicationBuilderExtensions.UseHealthChecks%2A>:
+* Terminates the pipeline when a request matches the health check endpoint. [Short-circuiting](xref:fundamentals/middleware/index) is often desirable because it avoids unnecessary work, such as logging and other middleware.
```

Comment:
Is this difference actually by design? To be specific, is it by design that `MapHealthChecks` does **not** short-circuit?
Both methods register the same `HealthCheckMiddleware`. The only difference is how it is mapped. See [here](https://github.com/dotnet/aspnetcore/blob/d47e49e9c1e173ac90821f7e89cc38e710274241/src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs#L77-L81) and [here](https://github.com/dotnet/aspnetcore/blob/d47e49e9c1e173ac90821f7e89cc38e710274241/src/Middleware/HealthChecks/src/Builder/HealthCheckApplicationBuilderExtensions.cs#L265).
Maybe related: https://github.com/aspnet/Diagnostics/pull/516

Reply:
The [routing documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/routing) doesn't explain it very clearly, but the [.NET 5.0 to .NET 6.0 migration document](https://learn.microsoft.com/en-us/aspnet/core/migration/50-to-60-samples?view=aspnetcore-7.0#:~:text=Routes%20added%20directly%20to%20the%20WebApplication%20execute%20at%20the%20end%20of%20the%20pipeline.) says:

> Routes [i.e. `MapGet()`] added directly to the [WebApplication](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.webapplication) execute at the _**end**_ of the pipeline.